reconcile/openshift_rolebindings.py

```diff
@@ -185,3 +196,31 @@ def run(dry_run, thread_pool_size=10, internal=None,
 
     if ri.has_error_registered():
         sys.exit(1)
+
+
+def has_valid_expiration_date(role):
+    date_bool = True
+    try:
+        role.get('expirationDate')
+    except AttributeError:
+        return date_bool
```

Comment:
The point of `.get` is not to have an exception. Either:

```python
try:
    role['expirationDate']
except KeyError:
    ...
```

or 
```python
if not role.get('expirationDate'):
    return date_bool
```

Given how graphql introduces the key even if it doesn't exist in app-interface, I'd recommend the latter.

Reply:
I was having a lot of trouble getting either of those two options to work, I have something similar to the first option that does work with my tests and with local q-r but let me know if it can be condensed down.